### ISSUE TITLE
[BUG]  A custom dropdown options provider is not working

### ISSUE THREAD

**Issue by jayachandra21** [elsa-workflows/elsa-core#6189]
## Description
 A custom dropdown options provider is not working.

## Steps to Reproduce
1. **Detailed Steps**:

- Class with VehicleUIHandler which inherits from **DropDownOptionsProviderBase**.

```
public class VehicleUIHandler : DropDownOptionsProviderBase
{
    private readonly Random _random = new();

    protected override ValueTask<ICollection<SelectListItem>> GetItemsAsync(PropertyInfo propertyInfo, object? context, CancellationToken cancellationToken)
    {
        var items = new List<SelectListItem>
        {
            new("BMW", "1"),
            new("Tesla", "2"),
            new("Peugeot", "3"),
            new(_random.Next(100).ToString(), "4")
        };

        return new(items);
    }
}

```
- Configures the specified property to refresh the UI when the property value changes.

````
public class RefreshUIHandler : IPropertyUIHandler
{
    public ValueTask<IDictionary<string, object>> GetUIPropertiesAsync(PropertyInfo propertyInfo, object? context, CancellationToken cancellationToken = default)
    {
        IDictionary<string, object> result = new Dictionary<string, object>
        {
            { "Refresh", true }
        };
        return ValueTask.FromResult(result);
    }
}
````

- Register in the UI handler.

`builder.Services.AddSingleton<VehicleUIHandler>();`

- Custom Activity **VehicleActivity**

````
public class VehicleActivity : Activity<string>
{
    [Input(
        Description = "The content type to use when sending the request.",
        UIHint = InputUIHints.DropDown,
        UIHandlers = [typeof(VehicleUIHandler), typeof(RefreshUIHandler)]
    )]
    public Input<string> Brand { get; set; } = default!;
}
````

- In the activity Input property is shows the empty list

![image](https://github.com/user-attachments/assets/d0ca01de-1fd0-45df-b0db-06392ad8dffa)

2. **Reproduction Rate**: every time

3. **Additional Configuration**:
   - Used the sample application as reference
   https://github.com/elsa-workflows/elsa-samples/tree/main/src/aspnet/Elsa.Samples.AspNet.CustomUIHandler

## Expected Behavior
The Vehicle activity suppose to show the list items mentioned in the VehicleUIHandler.

## Actual Behavior
It not listing the items. It means dropdown is empty.

## Environment
- **Elsa Package Version**: Elsa 3.3.0 RC3
- **Operating System**: Windows
- **Browser and Version**: Chrome

## Troubleshooting Attempts

- Tried to debug the VehicleUIHandler class but debugger is not hitting the same class.

**Comment by sfmskywalker** [comment]
Turned out that the sample project registered the handler incorrectly. Here's how to do it correctly:

```csharp
builder.Services.AddScoped<IPropertyUIHandler, VehicleUIHandler>();
builder.Services.AddScoped<IPropertyUIHandler, RefreshUIHandler>();
```

After doing that, it will work.
The sample project has been updated.

Thanks for reporting!

**Comment by Suchiman** [comment]
Huh indeed, that fixed it, in 3.1 it was not necessary to register them with DI at all but in 3.3 it doesn't work without it being registered.
